src/daemon.c

```diff
@@ -293,15 +291,22 @@ int become_daemon(int dont_fork, int close_all_files, const char *user, const ch
 		close(dev_null);
 
 	// generate our pid file
-	{
+	if(pidfile[0]) {
 		unlink(pidfile);
```

Comment:
1. I know that unlink can be done on opened file, but this should be performed in unique cases.
2. if you run as netdata and pid is on /run then the packaging will touch /run/netdata.pid && chown netdata /run/netdata.pid then execute your program. this means that if you remove this file, you won't be able to create it as /run permissions do not allow that. this is why you should truncate before write, and fallback to truncate when cleanup.
3. end result: you have valid pid within file regardless if you started as root or netdata, and when stop, either have pid file removed or size 0.